debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java

```diff
@@ -262,8 +245,62 @@ else if (oplogStart.getTime() > 0) {
                     }
                 }
 
+                if (taskContext.getConnectorConfig().isCursorReadPreferenceMonitoringEnabled() && preferenceRateLimiter.tryConsume(1)
+                        && !isSelectedReadPreference(client, mongo.getPreference(), cursor)) {
+                    LOGGER.info("Closing and recreating cursor due to election invalidating read preference");
+                    cursor.close();
+                    // Note: There is an edge case currently unaccounted for when an election happens during the first iteration of the cursor.
+                    // In this case, we need to set resume token in rsOffsetContext for this to skip the initial snapshot last timestamp.
+                    // But this is currently immutable.
+                    rsChangeStream = createChangeStream(client, rsOffsetContext);
```

Comment:
@jcechace  Will this _always_ result in a reevaluation of the read preference?

Reply:
I believe we discussed this yesterday and the answer was no?

Reply:
The answer is yes, but subject to the current constraints that the connection is "direct" and single node, thus it can never work like this until we can relax that constraint.

Reply:
Reworked this area, so resolving.